src/services/rowService.js

```diff
@@ -1,8 +1,14 @@
 module.exports = (db) => {
   return {
```

Comment:
Hmm, I think I know what's happening here. We seem to be allowing for some `data` elements not being present.  I think a neater solution would be to fill in the missing fields before calling `rowService` - then the logic here in `rowService` can stay precise, neat, and tidy.